packages/embark-ui/package.json

```diff
@@ -37,6 +37,9 @@
     "storybook": "start-storybook -p 9009 -s public",
     "test_": "node scripts/test.js"
   },
+  "dependencies": {
+    "date-fns": "1.30.1"
```

Comment:
The new dep can/should be moved to devDeps, since the build of embark-ui is completely static and can have zero deps in a production install.

Reply:
@michaeljyeates I've changed and updated the PR accordingly.

However, it's unclear to me what you mean by embark-ui being completely static and therefor **can** have zero deps.

Semantically (at least) the `date-fns` functions are a dependency of the application, not it's build process (or anything alike).

Would you mind elaborating what this is about?

Reply:
When running `yarn build` in `packages/embark-ui` or in the root, all JS that ends up in the production build is packed into it by webpack such that there are zero runtime dependencies that are loaded from `node_modules`. Having realized that's what CRA was/is doing for us, that's why I moved all of its deps into devDeps some time ago. We could list a dependency in deps, but it's pointless — the code will never be loaded from `node_modules` in a production contenxt.

Reply:
> can have zero deps

I meant simply that it's not necessary to put it deps, we can put it in devDeps.

Reply:
I think I understand. Yes when it comes to front-end code, usually deps are bundled with the application. But I haven't seen projects therefore moving their deps to devdeps.

We can keep it that way, it might be a point of confusion though (just like it confused me haha)

Reply:
Most of the time runtime deps are loaded from `node_modules`, so makes sense and is necessary to have them in deps.  But if they'll never be loaded from `node_modules` in production because prior to publishing the package all deps get packed into the `.js` being distributed, then having them as deps (instead of devDeps) just makes the production install heavier with no benefit to the user who installs it.

Does that make more sense?  Sorry for being unclear or confusing.  The same principle applies to DApps that are intended for deployment to ipfs / swarm. When they're deployed, it's all static files (and no `node_modules`). So those kinds of DApps only need devDeps, not deps.

Reply:
> all deps get packed into the .js being distributed, then having them as deps (instead of devDeps) just makes the production npm|yarn install heavier with no benefit to the user who installs it.

I might be wrong here as I'm certainly no expert when it comes to packaging and build tooling but aren't things like dead-code-elimination and tree-shaking there to help out with exactly that?

Usually with modern front-end tools the production build artifacts really only contain what's needed regardless of what's in node_modules/deps.

Not trying to be resistant here, just want to make sure I'm not missing anything.

Reply:
I'm sorry, I completely missed your latest reply, didn't mean to ignore it. This PR just got merged, but it still seems good to reply here.

Let's talk about this specific case, to avoid generalities that may be confusing the issue. `package.json` has some informative qualities but I'm focusing on what actually happens when the `embark-ui` package is installed in a production context.

Here are the points to consider relative to a production install:

1. If there are any packages listed in `embark-ui`'s `"dependencies": {...}` then they will be unconditionally installed in some `node_modules` folder, could be `embark-ui/node_modules` or auto-deduped by npm/yarn into a higher-up `node_modules`.

2. The scripts in `embark-ui/build` directory will never load modules from any `node_modules` because `embark-ui`'s build script (supplied by CRA), which we run as part of doing a release, packs all of the transpiled `embark-ui/src` code along with all dependencies (things loaded with `import` or `require` by `embark-ui/src` scripts) into the scripts in `embark-ui/build`.

Therefore, it doesn't make sense for `embark-ui`'s `package.json` to have `"dependencies": {...}`. They would always be installed but never loaded from `node_modules` in a production context. 

Consider that one of `embark-ui`'s dependencies is `monaco-editor`. The tarball for that is 11 MB, it's huge. That package is definitely used in a production install, *but* it's never loaded from any `node_modules`, rather it's optimized and packed into a `.js` file in `embark-ui/build` during release. So if we listed it in `"dependencies": {...}`, the only effect would be for a production install to download an additional 11 MB, with no benefit.

---

Coming at this from a different angle:

Having production runtime dependencies listed in `"dependencies": {...}` can be informative to users/developers — *"these are the deps used by a production install of this package".*

I definitely get that. However, it's important to consider other aspects of how the package is developed and what's included in a published tarball. In cases like `embark-ui`, where no production runtime code is loaded from `node_modules`, then having `"dependencies": {...}` is wasteful since unnneeded stuff gets pulled across the network and stored on disk.

I believe the benefits of reducing that waste outweigh the impact of `package.json` being less informative about which packages are used in development vs. production.